app/src/org/commcare/CommCareApplication.java

```diff
@@ -410,20 +410,27 @@ public void initializeGlobalResources(CommCareApp app) {
     }
 
     @NonNull
+    @Deprecated
     public String getPhoneId() {

```

Comment:
Not Sure about the use-case of this particular method, but it won't work with android 10.
Android best practices suggest [See 3rd option](https://developer.android.com/training/articles/user-data-ids#best-practices-android-identifiers) using either a random UUID(we'll have to save it in preferences to make sure it doesn't change) or Firebase Instance ID([read here](https://firebase.google.com/docs/reference/android/com/google/firebase/iid/FirebaseInstanceId)). 

Though both of them have a drawback: if application is reinstalled or user clears app data then the id will be lost and a new id will be generated. 

Let me know, if this is something important and we should use one of the above approaches to get a id.

Reply:
So `deviceId` was useful to identify the 2 separate installations of the app. I think using Firebase Instance Id seems like a good workaround here to separate out the installations as we don't have any other id currently that we maintain that's unique per CC installation.